services/llm-provider/src/adapters/bifrost_adapter.py

```
@@ -179,6 +196,10 @@ async def complete_chat(
                         )
                         result = self._parse_chat_response(response, use_model)
                     except PolicyGatewayBypassError:
+                        if self._requires_policy_gateway_mediation:
+                            raise RuntimeError(
+                                "policy gateway bypass is not allowed in production or release-gate mode"
+                            )
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)